modules/apps/message-boards/message-boards-web/src/main/resources/META-INF/resources/message_boards/js/MBPortlet.es.js

```diff
@@ -12,225 +12,243 @@
  * details.
  */
 
-import {PortletBase, fetch} from 'frontend-js-web';
-import core from 'metal';
-import {EventHandler} from 'metal-events';
+import {fetch} from 'frontend-js-web';
 
 const RECENTLY_REMOVED_ATTACHMENTS = {
 	multiple: Liferay.Language.get('x-recently-removed-attachments'),
 	single: Liferay.Language.get('x-recently-removed-attachment'),
 };
 
+const CONFIRM_DISCARD_IMAGES = Liferay.Language.get(
+	'uploads-are-in-progress-confirmation'
+);
+
 /**
  * MBPortlet handles the actions of replying or editing a
  * message board.
- *
- * @abstract
- * @extends {PortletBase}
  */
 
-class MBPortlet extends PortletBase {
-
-	/**
-	 * @inheritDoc
-	 */
-
-	created() {
-		this.eventHandler_ = new EventHandler();
+export default function MBPortlet({
+	constants,
+	currentAction,
+	getAttachmentsURL,
+	namespace,
+	replyToMessageId,
+	rootNodeId,
+	strings = {
+		confirmDiscardImages: CONFIRM_DISCARD_IMAGES,
+	},
+	viewTrashAttachmentsURL,
+}) {
+	const rootNode = document.getElementById(rootNodeId);
+
+	const workflowActionInputNode = rootNode.querySelector(
+		`#${namespace}workflowAction`
+	);
+
+	const publishButton = rootNode.querySelector(
+		'.button-holder button[type="submit"]'
+	);
+
+	if (publishButton) {
+		publishButton.addEventListener('click', () => {
+			workflowActionInputNode.value = constants.ACTION_PUBLISH;
+			saveFn();
+		});
 	}
 
-	/**
-	 * @inheritDoc
-	 */
-
-	attached() {
-		const publishButton = this.one('.button-holder button[type="submit"]');
+	const saveDrafButton = rootNode.querySelector(`#${namespace}saveButton`);
 
-		if (publishButton) {
-			this.eventHandler_.add(
-				publishButton.addEventListener('click', (e) => {
-					this.publish_(e);
-				})
-			);
-		}
+	if (saveDrafButton) {
+		saveDrafButton.addEventListener('click', () => {
+			workflowActionInputNode.value = constants.ACTION_SAVE_DRAFT;
+			saveFn();
+		});
+	}
 
-		const saveButton = this.one('#saveButton');
+	const advancedReplyLink = rootNode.querySelector('.advanced-reply');
 
-		if (saveButton) {
-			this.eventHandler_.add(
-				saveButton.addEventListener('click', (e) => {
-					this.saveDraft_(e);
-				})
-			);
-		}
-
-		const advancedReplyLink = this.one('.advanced-reply');
+	if (advancedReplyLink) {
+		advancedReplyLink.addEventListener('click', () => {
+			openAdvancedReply();
+		});
+	}
 
-		if (advancedReplyLink) {
-			this.eventHandler_.add(
-				advancedReplyLink.addEventListener('click', (e) => {
-					this.openAdvancedReply_(e);
-				})
-			);
-		}
+	const searchContainerId = `${namespace}messageAttachments`;
+	let _searchContainer;
```

Comment:
While totally fine, it feels a bit weird to hoist this and use the closure to access `_searchContainer` later on.

More than that, this exposes a race condition. It's not guaranteed that `_searchContainer` is properly set when used in the other contexts, so it would be safer to always use `Liferay.componentReady` or subordinate all other logic to it.